okhttp/src/jvmMain/kotlin/okhttp3/Dispatcher.kt

```diff
@@ -86,41 +86,49 @@ class Dispatcher() {
   @get:Synchronized
   var idleCallback: Runnable? = null
 
-  private var executorServiceOrNull: ExecutorService? = null
-
   @get:Synchronized
-  @get:JvmName("executorService") val executorService: ExecutorService
-    get() {
-      if (executorServiceOrNull == null) {
-        executorServiceOrNull = ThreadPoolExecutor(0, Int.MAX_VALUE, 60, TimeUnit.SECONDS,
-            SynchronousQueue(), threadFactory("$okHttpName Dispatcher", false))
-      }
-      return executorServiceOrNull!!
-    }
+  @set:Synchronized
+  @get:JvmName("priorityOrder")
+    /**
+     * A comparator that determines the priority of the async calls.
+     * A call with the highest priority will be executed first.
+     *
+     * When [priorityOrder] is null or returns 0, calls that are enqueued first have priority.
+     */
+  var priorityOrder: Comparator<Call>? = null
```

Comment:
@swankjesse Dispatcher is already a mutable object, this follows that pattern.  Do we want to continue that?

Could we deprecate the mutable features and move towards a Dispatcher Builder (in a follow up PR?)

Reply:
Yeah Dispatcher is mutable, and I’m not necessarily in a hurry to change that. I can imagine changing  `maxRequests` and `maxRequestsPerHost` dynamically in a running program.